eZ/Publish/Core/Search/Legacy/Content/IndexerGateway.php

```diff
@@ -0,0 +1,147 @@
+    }
+
+    public function getContentInSubtree(string $locationPath, int $iterationCount): Generator
+    {
+        $query = $this->buildQueryForContentInSubtree($locationPath);
+
+        yield from $this->fetchIteration($query->execute(), $iterationCount);
+    }
+
+    public function countContentInSubtree(string $locationPath): int
+    {
+        $query = $this->buildCountingQuery(
+            $this->buildQueryForContentInSubtree($locationPath)
+        );
+
+        return (int)$query->execute()->fetchOne();
+    }
+
+    public function getAllContent(int $iterationCount): Generator
+    {
+        $query = $this->buildQueryForAllContent();
+
+        yield from $this->fetchIteration($query->execute(), $iterationCount);
+    }
+
+    public function countAllContent(): int
+    {
+        $query = $this->buildCountingQuery(
+            $this->buildQueryForAllContent()
+        );
+
+        return (int)$query->execute()->fetchOne();
+    }
+
+    private function buildQueryForContentSince(DateTimeInterface $since): QueryBuilder
+    {
+        return $this->connection->createQueryBuilder()
+            ->select('c.id')
+            ->from('ezcontentobject', 'c')
+            ->where('c.status = :status')->andWhere('c.modified >= :since')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, ParameterType::INTEGER)
+            ->setParameter('since', $since->getTimestamp(), ParameterType::INTEGER);
+    }
+
+    private function buildQueryForContentInSubtree(string $locationPath): QueryBuilder
+    {
+        return $this->connection->createQueryBuilder()
+            ->select('DISTINCT c.id')
+            ->from('ezcontentobject', 'c')
+            ->innerJoin('c', 'ezcontentobject_tree', 't', 't.contentobject_id = c.id')
+            ->where('c.status = :status')
+            ->andWhere('t.path_string LIKE :path')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, ParameterType::INTEGER)
+            ->setParameter('path', $locationPath . '%', ParameterType::STRING);
+    }
+
+    private function buildQueryForAllContent(): QueryBuilder
+    {
+        return $this->connection->createQueryBuilder()
+            ->select('c.id')
+            ->from('ezcontentobject', 'c')
+            ->where('c.status = :status')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, ParameterType::INTEGER);
+    }
+
+    /**
+     * @throws \Doctrine\DBAL\Exception
+     */
+    private function buildCountingQuery(QueryBuilder $query): QueryBuilder
+    {
+        $databasePlatform = $this->connection->getDatabasePlatform();
+
+        // wrap existing select part in count expression
+        $query->select(
+            $databasePlatform->getCountExpression(
+                $query->getQueryPart('select')[0]
+            )
+        );
+
+        return $query;
+    }
+
+    private function fetchIteration(ResultStatement $stmt, int $iterationCount): Generator
```

Comment:
I would avoid name like $stmt and rather use full variable name $statement for readability.

Reply:
Fixed via 319e763a